src/domain/relay/limit-addresses.mapper.ts

```diff
@@ -318,19 +298,11 @@ export class LimitAddressesMapper {
       return false;
     }
 
-    const safeL1Deployment = getSafeSingletonDeployment({
-      version: args.version,
-      network: args.chainId,
-    });
-    const safeL2Deployment = getSafeL2SingletonDeployment({
-      version: args.version,
-      network: args.chainId,
-    });
+    const safeL1Deployments = getSafeSingletonDeployments(args);
+    const safeL2Deployments = getSafeL2SingletonDeployments(args);
 
-    const isL1Singleton =
-      safeL1Deployment?.networkAddresses[args.chainId] === singleton;
-    const isL2Singleton =
-      safeL2Deployment?.networkAddresses[args.chainId] === singleton;
+    const isL1Singleton = safeL1Deployments.includes(singleton);
+    const isL2Singleton = safeL2Deployments.includes(singleton);
```

Comment:
Nit: in this case, I think it could be also simplified, and optimized (potentially avoiding a second lookup):

```typescript
return (
      getSafeSingletonDeployments(args).includes(singleton) ||
      getSafeL2SingletonDeployments(args).includes(singleton)
    );
```

Wdyt?

Reply:
Agreed. Changed in 79fa30cc6d681b6b4f8ceaed1d063464d5ade42e.